### DISCUSSION TITLE
How to report out of date formula?

### DISCUSSION THREAD

**Discussion by barchard** [Homebrew/discussions#1876] [Everyday usage]
Hi, I'm looking for a guide to report an out of date formula. Are there any guidelines? 

Thanks

**Comment by nandahkrishna** [comment]
You could create a pull request to update the formula. Here are some docs to help you get started:
* [Contributing Guide](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
* [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook)
* [How to open a Homebrew PR](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request)

**Comment by SMillerDev** [comment]
The guidelines are here: https://docs.brew.sh/Updating-Software-in-Homebrew

**Comment by barchard** [comment]
Thanks but this assumes some technical knowledge. How about users who’d like to report them but can’t fix them themselves?

**Reply by carlocab** [reply]
We don't really provide an avenue for reporting outdated versions since it's generally very simple to update a formula's version.

The only knowledge it assumes is that you know how to use Git and GitHub. (Since you're here, I'll assume you're mostly ok with the latter.)

In fact, you could probably do away with the knowledge I mention above and just do
```
brew bump-formula-pr --version=<new-version> <outdated-formula>
```

You can also do it manually by updating the `url` and `sha256` in the outdated formula by doing
```
brew edit <outdated-formula>
```
You can then submit a PR with your changes as usual with the commit message
```
<name-of-outdated-formula> <new-version>
```
You can find your local copy of the Homebrew/core repository at
```
$(brew --repo homebrew/core)
```

If you get stuck with either of the above, feel free to ask for help here.

**Reply by iMichka** [reply]
Another solution would be to open a new discussion here asking if someone would be gentle enough to update it for you. 

If nobody cares and it’s not done right away, you could also even pay someone to do it for you, if it’s urgent.

It all boils down to the interest of a maintainer picking up that task, but in general we made our documentation and tooling as friendly as possible to actually let our users update versions themselves. There are more than 5000 packages in Homebrew and it would be insane if 30 maintainers had to keep them all up to date at any time.

**Reply by barchard** [reply]
Done, thanks. https://github.com/Homebrew/homebrew-core/pull/81801

**Comment by gromgit** [comment]
Before doing everything already mentioned, you should first [search for any open or closed PRs](https://github.com/Homebrew/homebrew-core/pulls) for the formula in question. Your target formula may already be in the process of being updated, or _can't_ be for some reason. For instance, there are several former core formulae that I'm now maintaining on [my macOS FUSE tap](https://github.com/gromgit/homebrew-fuse), that can't be upgraded because the macFUSE library they depend on is currently stuck on an older API version.
